source/slang/slang-parser.cpp

```diff
@@ -3360,6 +3360,18 @@ static Decl* ParseBufferBlockDecl(
         reflectionNameModifier->nameAndLoc = bufferVarDecl->nameAndLoc;
         parser->ReadToken(TokenType::Semicolon);
     }
+    else if (parser->options.allowGLSLInput &&
+             parser->LookAheadToken(TokenType::Identifier) &&
+             parser->LookAheadToken(TokenType::LBracket, 1) &&
+             parser->LookAheadToken(TokenType::RBracket, 2) &&
```

Comment:
I think the lookahead should stop at "[". We shouldn't need to ensure it is followed by "[];", as sized arrays should also be allowed.

Reply:
Since `parseBracketTypeSuffix` already looks for '[', you can just call it unconditionally.

Reply:
The logic here is to allow structs and other block declarations to end without ";".

If we want the disambiguation logic to be correct, we need to skip until we find the matching `]` and see if it is followed by ';'.

But since we are in GLSL compatbiliity mode (`parser->options.allowGLSLInput`), it is fine to just look for '[' and be loose about not allowing missing semicolons here.

Reply:
Okay, thanks for the review. Would it be better to just skip the `parser->ReadToken(TokenType::Semicolon)`? I left it in there for now.